src/fuzzer.rs

```diff
@@ -20,26 +21,40 @@ pub async fn new(urls: Vec<String>, browser: Browser, opt: parser::Options) {
 				Ok(res) => res,
 				Err(err) => {
 					detail = err.to_string();
-					return Ok((url, false, !is_err, detail))
+					return Ok((url, false, !is_err, detail, None))
 				}
 			};
 			let vuln: bool = match page.evaluate(CHECK_SCRIPT).await {
-				Ok(res) => {
-					page.close().await.unwrap();
-					res.into_value().unwrap()
-				},
+				Ok(res) => res.into_value().unwrap(),
 				Err(_) => false
 			};
 
 			Ok::<_, Box<dyn std::error::Error>>
-			((url, vuln, is_err, detail))
+			((url, vuln, is_err, detail, Some(page)))
 		}
 	)).buffer_unordered(opt.concurrency);
 
 	while let Some(res) = stream.next().await {
-		if let Ok((ref url, vuln, is_err, detail)) = res {
+		if let Ok((ref url, vuln, is_err, detail, page)) = res {
 			if vuln {
-				println!("[{}] {}", "VULN".green(), url)
+				println!("[{}] {}", "VULN".green(), url);
+
+				let mut target = Url::parse(url).unwrap();
+				target.set_query(None);
+
+				if let Some(ref p) = page {
+					let gadgets: Vec<String> = p
+						.evaluate(
+							include_str!("fingerprint.js")
+						)
+						.await
```

Comment:
If you're awaiting here this will block until the page is evaluated, meaning all the pending scans aren't polled in the meantime